cli/src/main/java/org/jboss/as/cli/operation/impl/DefaultOperationCandidatesProvider.java

```diff
@@ -336,6 +310,73 @@ public CommandLineCompleter getValueCompleter() {
         return result;
     }
 
+    private CommandLineCompleter getCompleter(final Map<String, CommandLineCompleterFactory> globalOpProps, final Property prop, CommandContext ctx, String operationName, OperationRequestAddress address) throws IllegalArgumentException {
+        CommandLineCompleter propCompleter = null;
+        final CommandLineCompleterFactory factory = globalOpProps == null ? null : globalOpProps.get(prop.getName());
+        if (factory != null) {
+            final String propName = ctx.getParsedCommandLine().getPropertyValue(Util.NAME);
+            if (Util.WRITE_ATTRIBUTE.equals(operationName) && propName != null) {
```

Comment:
I think this logic should be encapsulated within the CommandLineCompleterFactory created here:

https://github.com/wildfly/wildfly-core/blob/master/cli/src/main/java/org/jboss/as/cli/operation/impl/DefaultOperationCandidatesProvider.java#L384

It's the job of the CommandLineCompleterFactory to provide the completer.

@Gonthim / @aloubyansky WDYT?